### DISCUSSION TITLE
Is it normal that a UPXed binary issues a different Exit Code?

### DISCUSSION THREAD

**Discussion by JJKylee** [staxrip/staxrip#453] [Q&A] [answered]
Hi,

While trying @DJATOM's new [x265 build](https://github.com/DJATOM/x265-aMod/releases/download/3.4%2B55/x265-x64-v3.4+55-aMod-gcc10.2.1+opt.7z) to see if the [memory leak issue](https://github.com/staxrip/staxrip/issues/445) is gone, I experienced [this really weird phenomenon](https://github.com/DJATOM/x265-aMod/issues/4).

- UPXed `x265` ⇒ error in StaxRip
- un-UPXed but same `x265` ⇒ OK in StaxRip.

(But both run OK on CMD. 😓)

The error message is simple:

```
Video encoding returned error exit code: -1073740940 (0xC0000374)
```

Well, since I've never experienced this kind of error with all the previous UPXed binaries, I'm curious where this problem comes from.

Does anyone have an answer? Enlighten me, please.

**Comment by Dendraspis** [comment, marked as answer]
Such an exit code is often indicating a internal problem with conversions, assignments or so. You would expect it in an UPX-packed version, not in the *clean* one.
So I would also say, that this is a new problem/bug in the new build.

And you're not alone, what you should have seen yet: https://github.com/staxrip/staxrip/issues/445#issuecomment-761522604

**Reply by JJKylee** [reply]
Well, my first observation was wrong. The `x265` binary yields the **same error** _even after un-UPXing_!
(To make a cheap excuse, I was confused because I was testing with **4** different `x265` builds. 😋)
So my question itself was wrong. UPX is not the culprit, (sorry, UPX! 😓), and the _build itself_ seems to have something erroneous.
Like you said, @Dendraspis, it seems to be an internal problem, but it doesn't seem to be related to UPX.

And about the memory leak issue, yes, we need to hold on. I hoped @DJATOM's new build would have fixed this issue, but... you know, I ended up finding another issue. Let's wait and see.

**Reply by Dendraspis** [reply]
> but... you know, I ended up finding another issue. 

So business as usual.  😁

**Comment by JJKylee** [comment]
This issue was related to the _Vapoursynth reader_ employed by @DJATOM's mod. It's now fixed and he released a [new build](https://github.com/DJATOM/x265-aMod/releases/tag/3.4%2B56) just now.

Now it's time to test whether there's still memory leak. Well, I'm afraid this will take some time. 😋
